next/frontend/utils/amplify.ts

```diff
@@ -0,0 +1,244 @@
+    region: process.env.NEXT_PUBLIC_AWS_REGION,
+
+    // OPTIONAL - Amazon Cognito User Pool ID
+    userPoolId: process.env.NEXT_PUBLIC_COGNITO_USER_POOL_ID,
+
+    // OPTIONAL - Amazon Cognito Web Client ID (26-char alphanumeric string)
+    userPoolWebClientId: process.env.NEXT_PUBLIC_COGNITO_CLIENT_ID,
+
+    // OPTIONAL - Enforce user authentication prior to accessing AWS resources or not
+    mandatorySignIn: false,
+
+    // OPTIONAL - This is used when autoSignIn is enabled for Auth.signUp
+    // 'code' is used for Auth.confirmSignUp, 'link' is used for email link verification
+    signUpVerificationMethod: 'code',
+  },
+  ssr: true,
+})
+
+// TODO - could be better, currently used only after login, AccountStatus should be replaced or rewritten
+export const mapTierToStatus = (tier?: Tier): AccountStatus => {
+  switch (tier) {
+    case Tier.QUEUE_IDENTITY_CARD:
+      return AccountStatus.IdentityVerificationPending
+    case Tier.NOT_VERIFIED_IDENTITY_CARD:
+      return AccountStatus.IdentityVerificationFailed
+    case Tier.IDENTITY_CARD:
+    case Tier.EID:
+      return AccountStatus.IdentityVerificationSuccess
+    case Tier.NEW:
+      return AccountStatus.Idle
+    default:
+      return AccountStatus.IdentityVerificationRequired
+  }
+}
+
+export const getAccessToken = async () => {
+  try {
+    const session = await Auth.currentSession()
+    const jwtToken = session.getAccessToken().getJwtToken()
+    if (!jwtToken) throw new Error('no jwt token found in current session')
+    return jwtToken
+  } catch (error) {
+    logger.error('error getting access token - redirect to login', error)
+    window.location.assign(ROUTES.LOGIN)
+    throw error
+  }
+}
+
+export interface GetSSRCurrentAuth {
+  userData: UserData | null
+  isAuthenticated: boolean
+}
+
+// provides all the user data frontend might need as server side props
+// this way, FE can always access cognito data in a sync manner
+export const getSSRCurrentAuth = async (
+  req: GetServerSidePropsContext['req'],
+): Promise<GetSSRCurrentAuth> => {
+  const SSR = withSSRContext({ req })
+  let userData = null
+  let isAuthenticated = false
+  try {
+    const currentUser = await SSR.Auth.currentAuthenticatedUser()
+    userData = currentUser.attributes || null
+    isAuthenticated = !!currentUser
+    return { userData, isAuthenticated }
+  } catch (error) {
+    // TODO filter out errors because of unauthenticated users
+    logger.error('getServersideAuth error: ', error)
+  }
+  return { userData, isAuthenticated }
+}
+
+export const getSSRAccessToken = async (req: GetServerSidePropsContext['req']): Promise<string> => {
+  const SSR = withSSRContext({ req })
+  try {
+    const currentSession = await SSR.Auth.currentSession()
+    return (currentSession?.getAccessToken()?.getJwtToken() as string) || ''
+  } catch (error) {
+    // TODO filter out errors because of unauthenticated users
+    logger.error('getServersideAuth error: ', error)
+  }
+  return ''
```

Comment:
Maybe `null` is better?

Reply:
this allows me to be lazier with typing in some places :) as you always get type string and don't have to handle "string | null can't be assigned to prop which expects string"

but if this has any serious drawbacks you know of will gladly comply